src/storage/store/chunked_node_group.cpp

```diff
@@ -461,5 +464,36 @@ std::unique_ptr<ChunkedNodeGroup> ChunkedNodeGroup::deserialize(MemoryManager& m
     return chunkedGroup;
 }
 
+void ChunkedNodeGroup::setUnused(MemoryManager& mm) {
+    dataInUse = false;
+    mm.getBufferManager()->getSpillerOrSkip([&](auto& spiller) { spiller.addUnusedChunk(this); });
+}
+
+void ChunkedNodeGroup::loadFromDisk(MemoryManager& mm) {
+    mm.getBufferManager()->getSpillerOrSkip([&](auto& spiller) {
+        std::unique_lock lock{spillToDiskMutex};
+        // Prevent buffer manager from being able to spill this chunk to disk
+        spiller.clearUnusedChunk(this);
+        for (auto& chunk : chunks) {
+            chunk->loadFromDisk();
+        }
+        dataInUse = true;
+    });
+}
+
+uint64_t ChunkedNodeGroup::spillToDisk() {
+    KU_ASSERT(!dataInUse);
```

Comment:
Should L485 be after the lock?

Reply:
I think what's more important here is that it might be possible for a thread to try and spill this chunk to disk, but have it be loaded from disk and accessed between where its removed from the full group set and when its actually spilled. In which case, if we want to support concurrent spilling to disk (the other solution would be to call spillToDisk while the Spiller's lock is held), I think we'd want to wrap the inner part (after locking) in an `if (!dataInUse) { ... }` conditional, so that if its loaded by another thread before the lock is acquired it won't spill it.